src/adaptation/precond.jl

```diff
@@ -2,389 +2,227 @@
 #### Robust online (co-)variance estimators.
 ####
 
-## Variance estimator
+abstract type MassMatrixAdaptor <: AbstractAdaptor end
 
-abstract type VarEstimator{T} end
+initialize!(adaptor::T, n_adapts::Int) where {T<:MassMatrixAdaptor} = nothing
+finalize!(adaptor::T) where {T<:MassMatrixAdaptor} = nothing
```

Comment:
I'd suggest removing the static type parameters:

```suggestion
initialize!(::MassMatrixAdaptor, ::Int) = nothing
finalize!(::MassMatrixAdaptor) = nothing
```